dot/sync/bootstrap_syncer.go

```diff
@@ -26,14 +27,18 @@ func newBootstrapSyncer(blockState BlockState) *bootstrapSyncer {
 	}
 }
 
+var errNoWorker = errors.New("no worker")
+
 func (s *bootstrapSyncer) handleNewPeerState(ps *peerState) (*worker, error) {
 	head, err := s.blockState.BestBlockHeader()
 	if err != nil {
 		return nil, err
 	}
 
 	if ps.number.Cmp(head.Number) <= 0 {
-		return nil, nil
+		return nil, fmt.Errorf(
+			"%w: for head number %s and peer state number %s",
+			errNoWorker, head.Number, ps.number)
```

Comment:
please revert this, this case isn't an error

Reply:
It's just to satisfy the `nilnil` linter. It complains whenever we return an invalid result (`nil` worker) and a nil error.

I changed the caller code to check for this error instead of returning `nil, nil` here. But up to you, I can revert it as well.

Reply:
I feel like errors should only be returned when there's actually an error happening, in this case, the fact that there's no worker isn't an error, it's expected, as we don't need to do anything if receiving a peer state below our current head state. definitely up for debate though i guess lol

Reply:
Reverted back in https://github.com/ChainSafe/gossamer/pull/2072/commits/7cba22899d39e6e99ce0161c38798c07720c3c8d